### PR TITLE
Backgroundtimer

### PR DESCRIPTION
**Pull request by wewa00**
Update functionality which is discussed in #38.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -107,11 +107,22 @@ class App extends Component {
     if (this.hasWhitelistedScenes()) {
       this.requestScenesStatus();
     }
+    var intervalId = setInterval(this.timer, 5000);
```
Comment (t0mg) [inline src/App.js:110]: please use `const` (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) or just dont declare it and do eg
`this.setState({intervalId: setInterval(this.timer, 5000)});`

File: src/App.js [hunk 2]
```diff
@@ -107,11 +107,22 @@ class App extends Component {
...
     this.json.setEventHandler(() => {});
     this.mqtt.setEventHandler(() => {});
+    // use intervalId from the state to clear the interval
+    clearInterval(this.state.intervalId);
+  }
+
+  timer=()=>{
+    for (let i = 0; i < this.state.whitelist.length; i++) {
+      this.requestDeviceStatus(this.state.whitelist[i]);
+    }
```
Comment (t0mg) [inline src/App.js:125]: you can use [Array.forEach](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) here, eg `this.state.whitelist.forEach(device => this.requestDeviceStatus(device));`

**Review by t0mg — COMMENTED** [review]
Sorry it took me so long to get back to you, I was on vacation.

File: src/App.js (hunk 2 above)
Comment (wewa00) [reply src/App.js:125]: OK, will try this. But this is used several times in that way in the existing code. Maybe it should be changed there to (but not with this PullRequest)?

Comment (t0mg) [reply src/App.js:125]: It's just syntactic sugar, either way is fine. The latter is just more modern.

Comment (wewa00) [reply src/App.js]: done

File: src/App.js (hunk 1 above)
Comment (wewa00) [reply src/App.js]: done

**Comment by wewa00** [comment]
Please see my last commit. I have considered your suggestions.

**Pull request merged**